<a href="https://colab.research.google.com/github/davinavc/Data-Analyst/blob/main/E-Commerce%20using%20Machine%20Learning%20/E_Commerce_using_Machine_Learning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [26]:
#import library
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import confusion_matrix, classification_report

In [12]:
# Load data 'ecommerce_banner_promo.csv'
data = pd.read_csv('https://storage.googleapis.com/dqlab-dataset/pythonTutorial/ecommerce_banner_promo.csv')

In [ ]:
# Seting: matplotlib and seaborn
sns.set_style('whitegrid')
plt.style.use('fivethirtyeight')

In [ ]:
#1. Data explore using head(), info(), describe(), shape
print("\n[1] Data eksplorasi dengan head(), info(), describe(), shape")
print("Top 5 data:")
print(data.head())
print("Dataset information:")
print(data.info())
print("Dataset statistic descriptive:")
print(data.describe())
print("Dataset size:")
print(data.shape)

In [ ]:
#2. Exploratory data analysis by checking the correlation of each feature using the corr() function
print("\n[2] Exploratory data analysis by checking the correlation of each feature using the corr() function")
print(data.corr())

In [ ]:
#3. Exploratory data analysis by checking the distribution of labels using the groupby() and size() functions
print("\n[3] Exploratory data analysis by checking the distribution of labels using the groupby() and size() functions")
print(data.groupby(['Clicked on Ad']).size())

In [ ]:
#4. Exploratory data analysis with visualization
#4a. Visualizing the number of users divided into age ranges using a histogram (hist()) plot
plt.figure(figsize=(10, 5))
plt.hist(data['Age'], bins = data.Age.nunique())
plt.xlabel('Age')
plt.tight_layout()
plt.show()

In [ ]:
#4b. Using pairplot() from the seaborn (sns) module to depict relationships between each feature
plt.figure()
sns.pairplot(data)
plt.show()

 **Data Preparation**

In [ ]:
#5. Check missing value
print("\n[5] Check missing value")
print(data.isnull().sum().sum())

**Modelling**

In [ ]:
#6. Modeling using Logistic Regression, using an 80:20 ratio for training vs. testing
print("\n[6] Modeling using Logistic Regression, using an 80:20 ratio for training vs. testing")
#6a. Drop non-numerical (object type) features from X, as Logistic Regression requires numerical input. Also, drop the target/label and assign the target variable to y
X = data.drop(['Ad Topic Line','City','Country','Timestamp','Clicked on Ad'], axis = 1)
y = data['Clicked on Ad']

In [27]:
#6b. splitting the data
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 42)

In [28]:
#6c. Modelling
# Call the classifier
logreg = LogisticRegression()
# Fit the classifier to the training data
logreg = logreg.fit(X_train,y_train)
# Prediksi model
y_pred = logreg.predict(X_test)

In [ ]:
#6d. Model Performance Evaluation
print("Model Performance Evaluation:")
print("Training Accuracy :", logreg.score(X_train, y_train))
print("Testing Accuracy :", logreg.score(X_test, y_test))

**Evaluation**

In [31]:
#7. Print Confusion matrix dan classification report
print("\n[7] Print Confusion matrix dan classification report")

#apply confusion_matrix function to y_test and y_pred
print("Confusion matrix:")
cm = confusion_matrix(y_test, y_pred)
print(cm)

#apply classification_report function to y_test and y_pred
print("Classification report:")
cr = classification_report(y_test, y_pred)
print(cr)


[7] Print Confusion matrix dan classification report
Confusion matrix:
[[85  4]
 [16 95]]
Classification report:
              precision    recall  f1-score   support

           0       0.84      0.96      0.89        89
           1       0.96      0.86      0.90       111

    accuracy                           0.90       200
   macro avg       0.90      0.91      0.90       200
weighted avg       0.91      0.90      0.90       200



The model performs very well in predicting whether users will click on the website or not, as indicated by an accuracy score of 0.90. The dataset has a balanced number of labels (balanced class), allowing performance evaluation to use the accuracy metric.